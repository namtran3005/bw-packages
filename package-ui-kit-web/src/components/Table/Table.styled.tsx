import * as React from 'react';
import styled from 'styled-components';
import { Table as MuiTable } from '@material-ui/core';
import { TableProps as MuiTableProps } from '@material-ui/core/Table';
import { Spacing } from '@bitwala-cryptobank-squad/package-theme';

import { Root as TableRow } from '../TableRow';
import { Root as TableCell } from '../TableCell';

interface RootProps extends MuiTableProps {
  innerRef?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
  transparent?: boolean;
  variant?: 'normal' | 'dense';
}

const RootComponent: React.FunctionComponent<RootProps> = ({
  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
  transparent,
  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
  variant,
  ...cleanProps
}) => <MuiTable {...cleanProps} />;

const Root = styled(MuiTable as React.ComponentType<RootProps>)`
  &&& {
    ${TableRow} {
      height: ${({ variant }) =>
        variant === 'dense' ? 'auto' : `${Spacing.unit * 5}px`};
    }
    ${TableCell} {
      padding: ${Spacing.unit}px
        ${({ variant }) => (variant === 'dense' ? 0 : Spacing.unit * 2)}px;
    }
  }
`;

Root.displayName = 'Table';
RootComponent.displayName = 'TableComponent';

export { Root, RootProps, RootComponent };
