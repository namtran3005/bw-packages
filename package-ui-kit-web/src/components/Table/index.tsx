export {
  Table,
  TableProps,
  TableBody,
  TableBodyProps,
  TableCell,
  TableCellProps,
  TableFooter,
  TableFooterProps,
  TableHead,
  TableHeadProps,
  TablePagination,
  TablePaginationProps,
  TableRow,
  TableRowProps,
  TableSortLabel,
  TableSortLabelProps,
} from './Table';
export { Root, RootProps, RootComponent } from './Table.styled';
