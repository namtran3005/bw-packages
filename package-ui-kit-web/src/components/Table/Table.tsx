import * as React from 'react';

import { TableBody, TableBodyProps } from '../TableBody';
import { TableCell, TableCellProps } from '../TableCell';
import { TableFooter, TableFooterProps } from '../TableFooter';
import { TableHead, TableHeadProps } from '../TableHead';
import { TablePagination, TablePaginationProps } from '../TablePagination';
import { TableRow, TableRowProps } from '../TableRow';
import { TableSortLabel, TableSortLabelProps } from '../TableSortLabel';
import { Root, RootProps } from './Table.styled';

export {
  RootProps as TableProps,
  TableBody,
  TableBodyProps,
  TableCell,
  TableCellProps,
  TableFooter,
  TableFooterProps,
  TableHead,
  TableHeadProps,
  TablePagination,
  TablePaginationProps,
  TableRow,
  TableRowProps,
  TableSortLabel,
  TableSortLabelProps,
};

export class Table extends React.Component<RootProps> {
  public static Body = TableBody;
  public static Cell = TableCell;
  public static Footer = TableFooter;
  public static Head = TableHead;
  public static Pagination = TablePagination;
  public static Row = TableRow;
  public static SortLabel = TableSortLabel;

  public render(): JSX.Element {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    return <Root {...(this.props as any)} />;
  }
}
