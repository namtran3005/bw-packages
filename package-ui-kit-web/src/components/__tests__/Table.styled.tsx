/* eslint-disable @typescript-eslint/no-explicit-any */
import * as React from 'react';
import { shallow } from 'enzyme';
import { Root } from '../Table/Table.styled';

const props = {};

describe('Table', () => {
  it('should render table', () => {
    const wrapper = shallow(<Root {...(props as any)} />);
    expect(wrapper).toMatchSnapshot();
  });
});
