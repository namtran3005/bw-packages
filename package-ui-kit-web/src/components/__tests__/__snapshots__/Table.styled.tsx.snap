// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render table 1`] = `
.c0.c0.c0 .sc-gsTCUz {
  height: 40px;
}

.c0.c0.c0 .sc-bdfBwQ {
  padding: 8px 16px;
}

<WithStyles(ForwardRef(Table))
  className="c0"
/>
`;
